src/content/Shader.cpp

```diff
@@ -0,0 +1,51 @@
+#include "Shader.h"
+#include <utils/Utils.h>
+
+bgfx::ShaderHandle Shader::loadShader(const char* basePath, const char* _name)
+{
+    char filePath[512];
+
+    const char* shaderPath = "shaders/dx9/";
+
+    switch (bgfx::getRendererType())
+    {
+        case bgfx::RendererType::Direct3D11:
+        case bgfx::RendererType::Direct3D12:
+            shaderPath = "shaders/dx11/";
+            break;
+
+        case bgfx::RendererType::OpenGL:
+            shaderPath = "shaders/glsl/";
+            break;
+
+        case bgfx::RendererType::Metal:
+            shaderPath = "shaders/metal/";
+            break;
+
+        case bgfx::RendererType::OpenGLES:
+            shaderPath = "shaders/essl/";
+            break;
+
+        default:
+            break;
+    }
+
+    strcpy(filePath, basePath);
```

Comment:
why not strncpy and strncat?

Reply:
That is based on the bgfx sample framework. We should rewrite that :)